### PR TITLE
Enterprise Knowledge Graph Documentation in one swagger json file

### PR DESCRIPTION
**Pull request by tengr**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by anuchandy** [comment]
@tengr could you provide context here? the PR title is "Enterprise Knowledge Graph Documentation in one swagger json file", but I see a completely new swagger not just doc update.

**Comment by tengr** [comment]
> @tengr could you provide context here? the PR title is "Enterprise Knowledge Graph Documentation in one swagger json file", but I see a completely new swagger not just doc update.

I closed the previous [PR](https://github.com/Azure/azure-rest-api-specs/pull/6658). All the swagger json are merged and put in a single file in this PR. Since the previous PR hasn't been merged, the update is just adding one json file.

**Comment by nschonni** [comment]
If the words are correct, you can also add them to https://github.com/Azure/azure-rest-api-specs/blob/master/custom-words.txt to fix the spellcheck.

**Comment by tengr** [comment]
> If the words are correct, you can also add them to https://github.com/Azure/azure-rest-api-specs/blob/master/custom-words.txt to fix the spellcheck.

sent a [PR ](https://github.com/Azure/azure-rest-api-specs/pull/6775) for that

**Comment by nschonni** [comment]
OK, you could also include it in this one 😄

**Comment by tengr** [comment]
can anyone approve this PR?

**Comment by tengr** [comment]
Can someone please approve this PR ASAP?

**Comment by tengr** [comment]
could someone take a look and approve this?

**Comment by anuchandy** [comment]
@tengr Checked with a member of API review board, please set up a meeting with the review board, I sent you the contact details over teams

**Comment by anuchandy** [comment]
@tengr There is semantic validation [error](https://dev.azure.com/azure-sdk/public/_build/results?buildId=83488&view=logs&jobId=95dbed5b-845d-57ba-33d3-f6c5e69afcf4).

**Comment by tengr** [comment]
> @tengr There is semantic validation [error](https://dev.azure.com/azure-sdk/public/_build/results?buildId=83488&view=logs&jobId=95dbed5b-845d-57ba-33d3-f6c5e69afcf4).

All checks passing by now. Whitelisted a few words.

**Comment by anuchandy** [comment]
@tengr Did you get a chance to get it reviewed by AzureAPIBoard? 

\\ cc @johanste

**Comment by tengr** [comment]
> @tengr Did you get a chance to get it reviewed by AzureAPIBoard?
> 
> \ cc @johanste

Yes. We had a meeting with the AzureAPIBoard. I'm still working on updating this PR.

**Comment by yungezz** [comment]
hi @tengr any update?

**Comment by tengr** [comment]
> hi @tengr any update?

Just updated.

**Comment by anuchandy** [comment]
@tengr Let me know once you've final approval from API review board.

**Comment by yungezz** [comment]
HI @anuchandy , could you pls continue reviewing and push to end?

File: specification/EnterpriseKnowledgeGraph/data-plane/Microsoft.EnterpriseKnowledgeGraph/stable/2018-12-03/EnterpriseKnowledgeGraphSwagger.json [hunk 1]
```diff
@@ -0,0 +1,2001 @@
...
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "query",
+            "name": "service",
```
Comment (johanste) [inline specification/EnterpriseKnowledgeGraph/data-plane/Microsoft.EnterpriseKnowledgeGraph/stable/2018-12-03/EnterpriseKnowledgeGraphSwagger.json]: Wasn't this an internal-only parameter? Or am I misremembering?

File: specification/EnterpriseKnowledgeGraph/data-plane/Microsoft.EnterpriseKnowledgeGraph/stable/2018-12-03/EnterpriseKnowledgeGraphSwagger.json [hunk 2]
```diff
@@ -0,0 +1,2001 @@
...
+            "type": "integer"
+          },
+          "partialResultsOnTimeout": {
+            "type": "boolean"
+          },
+          "queryExecutionTime": {
+            "type": "string"
+          },
+          "isAggregate": {
+            "type": "boolean",
```
Comment (johanste) [inline specification/EnterpriseKnowledgeGraph/data-plane/Microsoft.EnterpriseKnowledgeGraph/stable/2018-12-03/EnterpriseKnowledgeGraphSwagger.json]: This is malformed JSON. I'm not really sure why this is not being flagged by our tools. But please fix this up.

**Comment by johanste** [comment]
The swagger document is malformed. Not sure why our tools are not flagging it. But it needs to be fixed up.

**Comment by tengr** [comment]
> The swagger document is malformed. Not sure why our tools are not flagging it. But it needs to be fixed up.

just fixed.

**Comment by johanste** [comment]
The semantic validation is now failing due to the document not being a valid swagger/openapi document.

File: specification/EnterpriseKnowledgeGraph/data-plane/Microsoft.EnterpriseKnowledgeGraph/stable/2018-12-03/EnterpriseKnowledgeGraphSwagger.json [hunk 3]
```diff
@@ -0,0 +1,2001 @@
...
+            "in": "body",
+            "name": "options",
+            "schema": {
+              "$ref": "#/definitions/SearchOptions"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
```
Comment (johanste) [inline specification/EnterpriseKnowledgeGraph/data-plane/Microsoft.EnterpriseKnowledgeGraph/stable/2018-12-03/EnterpriseKnowledgeGraphSwagger.json]: Will the number of items/the size of the data ever require the service to do paging?

File: specification/EnterpriseKnowledgeGraph/data-plane/Microsoft.EnterpriseKnowledgeGraph/stable/2018-12-03/EnterpriseKnowledgeGraphSwagger.json [hunk 4]
```diff
@@ -0,0 +1,2001 @@
...
+              "schema": {
+                "$ref": "#/definitions/NotFoundResult"
+              }
+            }
+          }
+        }
+      }
+    },
+    "definitions": {
+      "ConverseRequest": {
```
Comment (johanste) [inline specification/EnterpriseKnowledgeGraph/data-plane/Microsoft.EnterpriseKnowledgeGraph/stable/2018-12-03/EnterpriseKnowledgeGraphSwagger.json]: There are missing descriptions for almost all attributes. Understanding what values are allowed for attributes like `entityType` will be very challenging for users of the API.

File: specification/EnterpriseKnowledgeGraph/data-plane/Microsoft.EnterpriseKnowledgeGraph/stable/2018-12-03/EnterpriseKnowledgeGraphSwagger.json [hunk 5]
```diff
@@ -0,0 +1,2001 @@
...
+          },
+          "suggestionInterpretation": {
+            "$ref": "#/definitions/InterpretationResult"
+          },
+          "actionEntityId": {
+            "type": "string"
+          }
+        }
+      },
+      "EntityType": {
```
Comment (johanste) [inline specification/EnterpriseKnowledgeGraph/data-plane/Microsoft.EnterpriseKnowledgeGraph/stable/2018-12-03/EnterpriseKnowledgeGraphSwagger.json]: Why do you have a model item with only a single property? Is the intent for the number of attributes to grow over time? If so, should attributes named `entityType` be of type "/definitions/EntityType` or should the attribute name be `entityTypeName`?

File: specification/EnterpriseKnowledgeGraph/data-plane/Microsoft.EnterpriseKnowledgeGraph/stable/2018-12-03/EnterpriseKnowledgeGraphSwagger.json [hunk 6]
```diff
@@ -0,0 +1,2001 @@
...
+          },
+          "metadata": {
+            "type": "object",
+            "additionalProperties": {
+              "type": "object"
+            }
+          }
+        }
+      },
+      "SparqlQueryDbResponse": {
```
Comment (johanste) [inline specification/EnterpriseKnowledgeGraph/data-plane/Microsoft.EnterpriseKnowledgeGraph/stable/2018-12-03/EnterpriseKnowledgeGraphSwagger.json]: I would suggest naming the type in a way that makes it clear that this is a [SPARQL response object](https://www.w3.org/TR/2013/REC-sparql11-results-json-20130321/#json-result-object) and add a description that indicates that this is a standard format, not something that is unique to this service.

**Comment by anuchandy** [comment]
@tengr could you take a look at review comments?

**Pull request closed without merging**